### PR TITLE
Use MKCALENDAR to create calendar collections

### PR DESCRIPTION
**Pull request by hobarrera**
This adds support for calendar creation for radicale.

TBH, I'm not sure if this will work, and travis is the fastest way to test it.

### CODE REVIEW

File: vdirsyncer/storage/dav.py [hunk 1]
```diff
@@ -302,6 +302,30 @@ class CalDiscover(Discover):
...
+                <D:displayname>{}</D:displayname>
+                <C:supported-calendar-component-set>
+                    <C:comp name="VEVENT"/>
+                    <C:comp name="VTODO"/>
+                </C:supported-calendar-component-set>
+                </D:prop>
+            </D:set>
+        </C:mkcalendar>'''.format(
+            self._get_collection_from_url(url)
+        ).encode()
```
Comment (untitaker) [inline vdirsyncer/storage/dav.py]: Please use `utf-8` here.

Comment (hobarrera) [inline vdirsyncer/storage/dav.py]: That's the default: https://docs.python.org/3/library/stdtypes.html#str.encode

Comment (hobarrera) [inline vdirsyncer/storage/dav.py]: Or did you mean that you prefer it explicitly stated?

**Comment by hobarrera** [comment]
I've no idea why this failed so horribly. It's based on your own MKCALENDAR branch. :(

File: vdirsyncer/storage/dav.py (hunk 1 above)
Comment (untitaker) [inline vdirsyncer/storage/dav.py]: I didn't know that, but I think I prefer it to be explicit since I switch between Py2 and Py3 all the time.

**Comment by untitaker** [comment]
@hobarrera can we close this?

**Comment by hobarrera** [comment]
Yeah, it's a bit beyond me right now :(

**Pull request closed without merging**